pkg/analyze/archivereader.go

```diff
@@ -0,0 +1,111 @@
+package analyze
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	scyllav1listers "github.com/scylladb/scylla-operator/pkg/client/scylla/listers/scylla/v1"
+	"github.com/scylladb/scylla-operator/pkg/scheme"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func getIndexerForType(indexers map[reflect.Type]cache.Indexer, objType reflect.Type) cache.Indexer {
+	if indexer, exists := indexers[objType]; exists {
+		return indexer
+	}
+	// Create new indexer for this type if it doesn't exist
+	indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	indexers[objType] = indexer
+	return indexer
+}
+
+func isYamlFile(fsys fs.FS, path string, d fs.DirEntry) bool {
+	if d.IsDir() {
+		return false
+	}
+	extension := filepath.Ext(path)
+	if (extension == ".yaml") || (extension == ".yml") {
+		return true // Assume it is a valid yaml file
+	}
+	if extension == "" {
+		content, err := fs.ReadFile(fsys, path)
+		if err != nil {
+			return false
+		}
+		var unmarshalledContent interface{}
+		err = yaml.Unmarshal(content, &unmarshalledContent)
+		if err != nil {
+			return false
+		}
+		return true
+	}
+	return false
+}
```

Comment:
This will return false for files with a ".YML" extension, for example. Maybe just doing this:
```suggestion
func isYamlFile(fsys fs.FS, path string, d fs.DirEntry) bool {
	if d.IsDir() {
		return false
	}
	content, err := fs.ReadFile(fsys, path)
	if err != nil {
		return false
	}
	var unmarshalledContent interface{}
	err = yaml.Unmarshal(content, &unmarshalledContent)
	if err != nil {
		return false
	}
	return true
}
```
or moving this logic to the WalkDir clousure would be better

Reply:
fixed